### ISSUE TITLE
3.0 - edd_get_payment_status does not return abandoned statuses when return_label is false

### ISSUE THREAD

**Issue by davidsherlock** [awesomemotive/easy-digital-downloads#9181]
### Expected behavior
When calling `edd_get_payment_status` and passing in an order ID with an _abandoned_ status I expect to get the raw status, instead I get false. Other statuses appear normal/

### Actual behavior
It returns `false`.

**Comment by davidsherlock** [comment]
The issue is caused by this code:

```
$found_key = array_search( strtolower( $status ), $keys );
$status    = $found_key && array_key_exists( $found_key, $keys ) ? $keys[ $found_key ] : false;
```

I am not sure the reason behind using this, but a simple array_key_exists would work a lot better since it's an associative array for the statuses and all the statues match the database versions.

**Comment by robincornett** [comment]
Confirmed. This was introduced when we fixed #9026, but the `$found_key` check should be more strict. What's happening here is that the array key for `abandoned` is `0` and that can be construed as a falsey value, rather than an array key.
